olive/workflows/run/run.py

```diff
@@ -147,12 +147,26 @@ def run_engine(package_config: OlivePackageConfig, run_config: RunConfig, data_r
 
         AzureMLSystem.olive_config = run_config.to_json()
 
-    no_evaluation = (
+    # Register passes since we need to know whether they need to run on target
+    for pass_config in get_used_passes(run_config):
```

Comment:
https://stackoverflow.com/questions/1271320/resetting-generator-object-in-python

Instead of invoking multiple times of the get_used_passes, we can
1. change it to list so that all subsequent call to get_used_passes will be removed.
2. or use tee like the link said.

Reply:
used list. thanks!